### PR TITLE
fix(DB/SAI): Speaker Mar'grom

### LINKED ISSUE #3507: [Creature/DB]Speaker Mar'grom dont say anything in combat
##### CURRENT BEHAVIOUR:
when you find Speaker Mar'grom in Outland, combat with him, he will not say anything.

##### EXPECTED BLIZZLIKE BEHAVIOUR:
he says: My message of unity is meant for the Ogre clans, not for you, $R!  while in combat. (picture from NGA, thanks to author)
image

##### STEPS TO REPRODUCE THE PROBLEM:

1. .go c 151923/151924/151925/151926/151927  until find him.
2. combat with him.
3. apply SQL in EXTRA NOTES.
4. restart, relog, combat with him.

##### EXTRA NOTES:

```sql
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 18693;

DELETE FROM `smart_scripts` WHERE (`entryorguid` = 18693) AND (`source_type` = 0) AND (`id` IN (4));
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(18693, 0, 4, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Speaker Mar\'grom - In Combat - say 0');

DELETE FROM `creature_text` WHERE (`CreatureID` = 18693) AND (`GroupID` = 0);
INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `id`, `text`, `type`, `language`, `Probability`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
(18693, 0, 0, 'My message of unity is meant for the Ogre clans, not for you, $R!', 12, 0, 100, 19884, 0, 'Speaker Mar\'grom - In Combat - say 0');
```

##### AC HASH/COMMIT:
1910b4ab35bf6ecc65f02df3e06cba4b6805318d

##### OPERATING SYSTEM:
Windows 10

##### MODULES:
ahbot bgreward buffer beastmaster trangmog

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/93265804-creature-db-speaker-mar-grom-dont-say-anything-in-combat?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Si1ker**
This PR will fix the following NPC=
ID: 18693
Name: Speaker Mar'grom

Reference for combat and dialogues are taken from: https://www.wowhead.com/npc=18693/speaker-margrom#comments
image

## CHANGES PROPOSED:
-  Add 4 different quotes when in combat
-  Add 6 (unused) dialogues when passive/moving
-  Improve SAI combat handling

## ISSUES ADDRESSED:
- Closes #3507 

## TESTS PERFORMED:
- Applied SQL
- Tested in-game

## HOW TO TEST THE CHANGES:

**Before applying SQL changes**
- .go c 151923 / 151924 / 151925 / 151926 / 151927 until finding him.
- .gm off
- Stand around him without any combat: You will notice there is no dialogue while he is around
- Enter in combat with him: You will also notice he does not say something when in combat

**After applying SQL changes**
- Test 1: Stand around him (around 20-30 yards should be enough), you will see different speeches being said for his comrades around
- Test 2: Enter in combat with him, he will say a quote
- Test 3: He has 4 spells, a fire-resistance buff (after any damage), a fireball (when at high HP), a frost ball (at medium-low HP), and a small fire spell (when he reaches low HP)
- 🐱 works!

## KNOWN ISSUES AND TODO LIST:

- Nothing to do here from now on, the NPC is 100% completed

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Remberance** [comment]
Thank you for perfecting this, i will learn how to make sai more perfect contimue.

**Review by FrancescoBorzi — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1601240498542654700.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+(18693, 1, 2, 'I decree that you are an enemy of the Ogre clans. Punishment is death!', 12, 0, 25, 19886, 0, 'Speaker Mar\'grom - Say aggro 3'),
+(18693, 1, 3, 'The Ogre clans will not be subjugated. Not by the gronn, and not by a puny $R like you!', 12, 0, 25, 21024, 0, 'Speaker Mar\'grom - Say aggro 4');
+
+DELETE FROM `smart_scripts` WHERE  `entryorguid`=18693 AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES	
+-- Speech (Passive)
+('18693', '0', '0', '0', '1', '0', '100', '0', '15000', '30000', '40000', '60000', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', 'Speaker Mar\'grom - In Combat - Say Speech'),
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1601240498542654700.sql:26]: you could avoid to use strings when numbers are expected

**Review by Helias — CHANGES_REQUESTED** [review]

**Comment by Si1ker** [comment]
@FrancescoBorzi @Helias  Done, check again.

**Review by Helias — APPROVED** [review]

**Comment by Remberance** [comment]
image
 worked

**Comment by FrancescoBorzi** [comment]
good job @Si1ker and thanks @Remberance for testing!

**Review by Helias — APPROVED** [review]

**Pull request merged**